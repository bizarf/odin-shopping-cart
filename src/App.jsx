import Navbar from "./components/Navbar";
import Home from "./components/Home";
import Shop from "./components/Shop";
import Cart from "./components/Cart";
import { useState } from "react";
import { BrowserRouter, Routes, Route } from "react-router-dom";

const App = () => {
    const [cart, setCart] = useState([]);

    const addToCart = (productId, productName, image, price) => {
        const item = {
            productId: productId,
            productName: productName,
            image: image,
            price: price,
            quantity: 1,
        };
        setCart([...cart, item]);
    };

    const removeItemFromCart = (productId) => {
        const cartIndex = cart.findIndex((id) => {
            return id.productId === productId;
        });
        const updatedCart = cart;
        updatedCart.splice(cartIndex, 1);
        console.log(updatedCart);
        setCart([...updatedCart]);
    };

    const increaseQuantity = () => {};

    const decreaseQuantity = () => {};

    return (
        <div>
            <BrowserRouter>
                <Navbar cart={cart} />
                <Routes>
                    <Route path="/" element={<Home />} />
                    <Route
                        path="/Shop"
                        element={
                            <Shop
                                setCart={setCart}
                                cart={cart}
                                addToCart={addToCart}
                            />
                        }
                    />
                    <Route
                        path="/Cart"
                        element={
                            <Cart
                                cart={cart}
                                removeItemFromCart={removeItemFromCart}
                                increaseQuantity={increaseQuantity}
                                decreaseQuantity={decreaseQuantity}
                            />
                        }
                    />
                </Routes>
            </BrowserRouter>
        </div>
    );
};

export default App;
