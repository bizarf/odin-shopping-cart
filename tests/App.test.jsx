import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import App from "../src/App";
import { describe, expect } from "vitest";

describe("App tests", () => {
    it("The app opens onto the homepage", () => {
        const { app } = render(<App />);

        expect(app).toMatchSnapshot();
    });
});
