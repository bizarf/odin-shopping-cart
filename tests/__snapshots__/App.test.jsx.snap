// Vitest Snapshot v1

exports[`App tests > The app opens onto the homepage 1`] = `undefined`;

exports[`Navbar tests > The Navbar renders 1`] = `undefined`;
